_pytest/unittest.py

```diff
@@ -157,9 +156,20 @@ def runtest(self):
             self._testcase(result=self)
         else:
             # disables tearDown and cleanups for post mortem debugging (see #1890)
+            # but still implements the skipping machinery (see #2137)
+            testMethod = getattr(self._testcase, self._testcase._testMethodName)
```

Comment:
This looks correct according to the implementation in [`unittest`](https://github.com/python/cpython/blob/master/Lib/unittest/case.py#L576), thanks!

Could you please move this entire logic to a separate method though? Something like:

```python
...
else:
    if self._handle_skip():
        return
    self._testcase.debug()
```

This makes the code easier to follow IMHO.